src/ui/buttons/props.js

```diff
@@ -765,18 +764,22 @@ export function normalizeButtonMessage(
 
   if (typeof offer !== "undefined") {
     if (!Array.isArray(offer)) {
-      throw new TypeError(
-        `Expected message.offer to be an array of strings, got: ${String(
-          offer
-        )}`
-      );
+      if (typeof offer !== "string") {
+        throw new TypeError(
+          `Expected message.offer to be an array of strings, got: ${String(
+            offer
+          )}`
+        );
+      }
+      offer = offer.split(",");
```

Comment:
Would the following help improve clarity of these different `if` blocks by removing a level of nesting and tying the conversion from string to array code closer to its respective `if` code?

``` js
if (typeof offer !== "undefined") {
    if (typeof offer === "string") {
        offer = offer.split(",");
    }
    if (!Array.isArray(offer)) {
        throw new TypeError(/* ... */)
    }

    // ...
}
```